src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3107,7 +3313,16 @@ void EmitAtomicSingleCharZeroOrOne(RegexNode node)
                 Debug.Assert(node.Kind is RegexNodeKind.Oneloop or RegexNodeKind.Oneloopatomic or RegexNodeKind.Notoneloop or RegexNodeKind.Notoneloopatomic or RegexNodeKind.Setloop or RegexNodeKind.Setloopatomic, $"Unexpected type: {node.Kind}");
                 Debug.Assert(node.M == 0 && node.N == 1);
 
-                string expr = $"{sliceSpan}[{sliceStaticPos}]";
+                bool rtl = (node.Options & RegexOptions.RightToLeft) != 0;
+                if (rtl)
+                {
+                    TransferSliceStaticPosToPos(); // we don't use static pos for rtl
+                }
+
+                string expr = !rtl ?
+                    $"{sliceSpan}[{sliceStaticPos}]" :
+                    "inputSpan[pos - 1]";
```

Comment:
> inputSpan

NIT: Should we consider to have this saved as a label instead, in order to avoid issues if for some reason we rename the parameter in the future?

Reply:
We hard code "inputSpan" and "pos" everywhere already.  You mean changing it everywhere?  Or something specific to this use?

Reply:
No need to do it now, I only brought it up here since you modified this particular line but I understand this is not the only instance of it, so we can do this later instead if we feel it would be good.

Reply:
Happy to extract it or see it extracted separately.